compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -901,11 +901,16 @@ impl<'block, Registers: RegisterAllocator> BrilligBlock<'block, Registers> {
                 // and become usize::MAX, and then return to 1, then it will indicate
                 // an array as mutable when it probably shouldn't be.
                 if self.brillig_context.enable_debug_assertions() {
-                    let min_addr = ReservedRegisters::usize_one();
+                    let zero =
+                        SingleAddrVariable::new(self.brillig_context.allocate_register(), 32);
+
+                    self.brillig_context.const_instruction(zero, FieldElement::zero());
```

Comment:
I know almost nothing about brillig so feel free to ignore this, but why the old and new code are the same? Or what's the difference?

Reply:
The old and new code aren't the same here, it was changed from `1 < rc` to `0 < rc` but I should probably revert this change since it was an earlier version of https://github.com/noir-lang/noir/pull/7849 before I broke it out into a new PR.

Reply:
Thanks!

Reply:
Ah, right one of the reference tests fails without this change. I'll merge the other PR into this one for now then.